lib/BackendClient.js

```diff
@@ -237,29 +237,31 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
-    function registerEntryInDatabase(entryObject, state, event) {
+    function registerEntryInDatabase(entryObj, state, event) {
```

Comment:
separated the call to registerEntryInDatabase from craeteRecordedEntry (new method)